apptest/tests/metricsql_test.go

```diff
@@ -173,3 +178,29 @@ func testInstantQueryDoesNotReturnStaleNaNs(t *testing.T, sut apptest.Prometheus
 		t.Errorf("unexpected response (-want, +got):\n%s", diff)
 	}
 }
+
+func testQueryRangeWithAtModifier(t *testing.T, sut apptest.PrometheusWriteQuerier) {
```

Comment:
This test do not panic on master branch

Reply:
Any chance you did not rebuild binary before running test from master?
Just tried on fresh master branch and can reproduce panic there with this test.

Reply:
Just discovered  that conversation of `NaN` to int64 is also platform-dependant, so value of `int64(math.NaN() * 1000)` can produce 0 or max int64 depending on platform implementation.